Funky-Chickens/Bangazon-CLI
# Jmr/El/Gm testusers

------------------el/gm
converted all the forEach insert functions into Promise.all, so that the async functions of building the database and running tests can be managed more effectively.
moved GET test into separate file to keep modularized.
wrote tests for GET and POST user.
Unfinished, passed code to team for continued exploration --
picked up by JMR

-------------------jmr
added a global before() statement that tells mocha to run functions contained within it before any tests in any file. the table creator function resolves *after* the last serialized db.run is finished and the rows are inserted.

To test: 
-Run "npm test"
-save somewhere in the file so it runs it's auto-watch script.
-^C and run npm test again
-^C and manually delete database (make sure it is unlocked so not open in db browser and the node testing isn't 'watching') and run npm test again
